app/lib/search/token_index.dart

```diff
@@ -97,32 +97,22 @@ class Score {
     if (threshold == null) {
       return this;
     }
-    final result = <String, double>{};
-    for (String key in _values.keys) {
-      final value = _values[key]!;
-      if (value < threshold) continue;
-      result[key] = value;
-    }
+    final result = Map<String, double>.fromEntries(
+        _values.entries.where((entry) => entry.value >= threshold!));
```

Comment:
The `threshold!` null check assertion shouldn't be necessary.
We could `final t = threshold!` on the line before, if the analyzer can't get it from `if(threshold == null) return this`.